Mergin/version_viewer_dialog.py

```diff
@@ -0,0 +1,588 @@
+
+        height = 30
+        self.toolbar.setMinimumHeight(height)
+
+        self.history_control.setMinimumHeight(height)
+        self.history_control.setVisible(False)
+
+        self.toggle_layers_action = QAction(
+            QgsApplication.getThemeIcon("/mActionAddLayer.svg"), "Toggle Project Layers", self
+        )
+        self.toggle_layers_action.setCheckable(True)
+        self.toggle_layers_action.setChecked(True)
+        self.toggle_layers_action.toggled.connect(self.toggle_project_layers)
+        self.toolbar.addAction(self.toggle_layers_action)
+
+        self.toolbar.addSeparator()
+
+        self.zoom_full_action = QAction(QgsApplication.getThemeIcon("/mActionZoomFullExtent.svg"), "Zoom Full", self)
+        self.toolbar.addAction(self.zoom_full_action)
+
+        self.zoom_selected_action = QAction(
+            QgsApplication.getThemeIcon("/mActionZoomToSelected.svg"), "Zoom To Selection", self
+        )
+        self.toolbar.addAction(self.zoom_selected_action)
+
+        btn_add_changes = QPushButton("Add to project")
+        btn_add_changes.setIcon(QgsApplication.getThemeIcon("/mActionAdd.svg"))
+        menu = QMenu()
+        add_current_action = menu.addAction(QIcon(icon_path("file-plus.svg")), "Add current changes layer to project")
+        add_current_action.triggered.connect(self.add_current_to_project)
+        add_all_action = menu.addAction(QIcon(icon_path("folder-plus.svg")), "Add all changes layers to project")
+        add_all_action.triggered.connect(self.add_all_to_project)
+        btn_add_changes.setMenu(menu)
+
+        self.toolbar.addWidget(btn_add_changes)
+        self.toolbar.setIconSize(iface.iconSize())
+
+        self.map_canvas.enableAntiAliasing(True)
+        self.map_canvas.setSelectionColor(QColor(Qt.cyan))
+        self.pan_tool = QgsMapToolPan(self.map_canvas)
+        self.map_canvas.setMapTool(self.pan_tool)
+
+        self.current_diff = None
+        self.diff_layers = []
+        self.filter_model = None
+        self.layer_list.currentRowChanged.connect(self.diff_layer_changed)
+
+        self.icons = {
+            "added": "plus.svg",
+            "removed": "trash.svg",
+            "updated": "pencil.svg",
+            "renamed": "pencil.svg",
+            "table": "table.svg",
+        }
+        self.model_detail = QStandardItemModel()
+        self.model_detail.setHorizontalHeaderLabels(["Details"])
+        self.details_treeview.setModel(self.model_detail)
+
+        self.versionModel.current_version = self.mp.version()
+
+    def exec(self):
+        try:
+            ws_id = self.mp.workspace_id()
+        except ClientError as e:
+            QMessageBox.warning(None, "Client Error", str(e))
+            return
+
+        # check if user has permissions
+        usage = self.mc.workspace_usage(ws_id)
+        if not usage["view_history"]["allowed"]:
+            QMessageBox.warning(None, "Permission Error", "The workspace does not allow to view project history.")
```

Comment:
How about something like

```suggestion
            QMessageBox.warning(None, "Upgrade required", "To view the project history, please upgrade your subscription plan.")
```

to let users know what to do to see the history? 
Optionally also with a button to navigate to the subscription page :)